client/app/location/location.js

```diff
@@ -44,43 +15,152 @@ angular.module('dinnerDaddy.location', [])
 
   var info = new google.maps.InfoWindow();
 
-/* 
-The plan is to have a marker type specific to users, and a marker type for the designated restaurant
-In a given session, each user will have unique coordinates and in the success function,
-the coordinates will be fed into the server socket. The deployed version will give a shared socket,
-so the coordinates for all group members can bubble up from server to each client */
+  var distanceMatrixService = new google.maps.DistanceMatrixService();
+  var directionService = new google.maps.DirectionsService();
+  var directionRenderer = new google.maps.DirectionsRenderer({preserveViewport: true, suppressMarkers: true});
+  directionRenderer.setMap(map);
+
+  /* --- Test data ---- !!!!
+   PROVIDE RESTAURANT INFO LIKE THIS !!! */
+   //will not accept location without coordinates
+  var restaurant = {
+    name: 'Golden Boy Pizza',
+    location: [37.79983540000001, -122.4080836]
+  };
+  /* ---- END ---- */
+
+  $scope.origin;
+  $scope.destination;
+  $scope.restaurantName = restaurant.name;
+  $scope.restaurantLocation = restaurant.location;
+  $scope.username = $cookies.get('name');
+  $scope.transport = google.maps.TravelMode.DRIVING;
+
+  $scope.data = {
+      availableOptions: [
+        {id: 'driving', name: 'DRIVING'},
+        {id: 'walking', name: 'WALKING'}
+      ],
+    selectedOption: {id: 'driving', name: 'DRIVING'} //This sets the default value of the select in the ui
+  };
+
+  $scope.updateMode = function () {
+    $scope.transport = google.maps.TravelMode[$scope.data.selectedOption.name];
+    LocationFactory.getDistance(origin, $scope.restaurantLocation, $scope.transport).then(function (distances) {
+      $scope.distance = distances.distance.text;
+      $scope.duration = distances.duration.text;
+    });
+    showRoutes();
+  };
 
+  var showRoutes = function () {
+    var routeQuery = {
+      origin: $scope.origin,
+      destination: $scope.destination,
+      travelMode: $scope.transport,
+      unitSystem: google.maps.UnitSystem.IMPERIAL
+    };
+
+    directionService.route(routeQuery, function (result, status) {
+      if (status === google.maps.DirectionsStatus.OK) {
+        directionRenderer.setDirections(result);
+        bounds = new google.maps.LatLngBounds();
+        bounds.extend(result.routes[0].overview_path[0]);
+        var k = result.routes[0].overview_path.length;
+        bounds.extend(result.routes[0].overview_path[k-1]);
+        panning = true;
+        map.panTo(bounds.getCenter());  
+      }
+    });
+  };
+
+  //position is fed in from google
   var success = function (position, username) {
     //gathering coordinates from user geolocation
-    var coords = new google.maps.LatLng(position.coords.latitude, position.coords.longitude);
-    
+    var coords = new google.maps.LatLng(position.coords.latitude, position.coords.longitude)
+    origin.push([position.coords.latitude, position.coords.longitude].join(','))
```

Comment:
Can you add a comment that this refers to the closure scope origin variable above?